### PR TITLE
`flutter start` initializes the Xcode project if the user has not already done so.

### PR DESCRIPTION
**Pull request by chinmaygarde**
cc @devoncarew. I have also moved the guts of the initialization logic out of the command file so that it can be called from multiple places.

### CODE REVIEW

**Comment by eseidelGoogle** [comment]
This sgtm, but can we just run ios --init as a part of create?  Or does ios --init require bits of xcode which create doesn't?

File: packages/flutter_tools/lib/src/commands/ios.dart [hunk 1]
```diff
@@ -178,165 +25,9 @@ class IOSCommand extends FlutterCommand {
     }
 
     if (argResults['init'])
-      return await _runInitCommand();
+      return await InitializeXcodeProjectHarness();
```
Comment (abarth) [inline packages/flutter_tools/lib/src/commands/ios.dart]: Lower-case "i"

**Comment by eseidelGoogle** [comment]
`flutter ios --init` made a lot of sense when the ios port was brand new, but doesn't make much sense now when we expect all devs will want both android and ios.  I could imagine wanting to have some sort of ios upgrade, but presumably that would just be rolled into `flutter upgrade`?

File: packages/flutter_tools/lib/src/ios/device_ios.dart [hunk 2]
```diff
@@ -560,8 +561,11 @@ String _getIOSEngineRevision(ApplicationPackage app) {
 
 Future<bool> _buildIOSXcodeProject(ApplicationPackage app, { bool buildForDevice }) async {
   if (!FileSystemEntity.isDirectorySync(app.localPath)) {
-    printError('Path "${path.absolute(app.localPath)}" does not exist.\nDid you run `flutter ios --init`?');
-    return false;
+    printTrace('Path "${path.absolute(app.localPath)}" does not exist. Initializing the Xcode project on behalf of the user.');
```
Comment (abarth) [inline packages/flutter_tools/lib/src/ios/device_ios.dart]: I'd remove "on behalf of the user".  Everything we do is on behalf of the user. :)

File: packages/flutter_tools/lib/src/ios/initialize_xcode.dart [hunk 3]
```diff
@@ -0,0 +1,316 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import "dart:async";
+import "dart:io";
+import "package:path/path.dart" as path;
```
Comment (abarth) [inline packages/flutter_tools/lib/src/ios/initialize_xcode.dart]: Blank line after `dart:` imports.  Also blank line after `package:` imports.

Comment (abarth) [inline packages/flutter_tools/lib/src/ios/initialize_xcode.dart]: Also, we use single quote `'` everywhere we can, including imports.

**Comment by abarth** [comment]
LGTM

File: packages/flutter_tools/lib/src/ios/initialize_xcode.dart (hunk 3 above)
Comment (chinmaygarde) [inline packages/flutter_tools/lib/src/ios/initialize_xcode.dart]: Ok. I just moved this code from one file to another. Will update.

**Comment by chinmaygarde** [comment]
> Or does ios --init require bits of xcode which create doesn't?

Yeah, create only initializes the user editable components of the Xcode project. It does not download all the device runners.

File: packages/flutter_tools/lib/src/ios/device_ios.dart (hunk 2 above)
Comment (chinmaygarde) [inline packages/flutter_tools/lib/src/ios/device_ios.dart]: >  Everything we do is on behalf of the user. :)

word https://www.youtube.com/watch?v=2QFg9CvgIYI

**Comment by chinmaygarde** [comment]
Addressed all comments and rebased.

**Comment by devoncarew** [comment]
lgtm, +1 to refactoring out of the command file.

nit re: the file naming, `initialize_xcode.dart` sounds like it's setting up xcode, not the app's xcode project. I don't have a better name off-hand.

As mentioned in this PR, I think we do want to move to doing things in `flutter create` and `flutter upgrade` like in the android workflow. Perhaps file a bug to capture that?

**Comment by chinmaygarde** [comment]
Fixes https://github.com/flutter/flutter/issues/1909

**Comment by chinmaygarde** [comment]
@devoncarew Just saw your comment updated inline. `setup_xcodeproj.dart` maybe then? Will update it in another patch.

**Comment by devoncarew** [comment]
setup_xcodeproj.dart sgtm

**Pull request merged**